File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Python_Solutions/Problem9.py
def newton_raphson(x0, tolerance=0.0001, max_iterations=10):
    x = x0
    print(f"Newton-Raphson Method for f(x) = x⁵ - 3x + 1 = 0")
    print(f"Initial guess: x₀ = {x0}")
    print("\nIterations:")
    
    for i in range(max_iterations):
        fx = x**5 - 3*x + 1  # f(x) = x⁵ - 3x + 1
        fpx = 5*x**4 - 3     # f'(x) = 5x⁴ - 3
        
        x_next = x - fx / fpx
        
        print(f"Iteration {i+1}: x = {x:.6f}, f(x) = {fx:.6f}")
        
        if abs(x_next - x) < tolerance:
            print(f"\nRoot found: {x_next:.6f}")
            break
            
        x = x_next

if __name__ == "__main__":
    newton_raphson(1.0)  # Initial guess 

File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Java_Solutions/Problem9.java
public class Problem9 {
    public static void main(String[] args) {
        double x0 = 1.0;  // Initial guess
        double tolerance = 0.0001;
        int maxIterations = 10;
        
        System.out.println("Newton-Raphson Method for f(x) = x⁵ - 3x + 1 = 0");
        System.out.println("Initial guess: x₀ = " + x0);
        System.out.println("\nIterations:");
        
        double x = x0;
        for (int i = 0; i < maxIterations; i++) {
            double fx = Math.pow(x, 5) - 3 * x + 1;  // f(x) = x⁵ - 3x + 1
            double fpx = 5 * Math.pow(x, 4) - 3;     // f'(x) = 5x⁴ - 3
            
            double xNext = x - fx / fpx;
            
            System.out.printf("Iteration %d: x = %.6f, f(x) = %.6f\n", 
                            i + 1, x, fx);
            
            if (Math.abs(xNext - x) < tolerance) {
                System.out.printf("\nRoot found: %.6f\n", xNext);
                break;
            }
            
            x = xNext;
        }
    }
} 